### ISSUE TITLE
[Terminal]: Screen Reader does not announce Pop-up name when its focus first jump inside pop-up.

### ISSUE THREAD

**Issue by v-rpundir** [microsoft/terminal#12054]
### Windows Terminal version

1.12.3472.0

### Windows build number

10.0.22504.1010

### Other Software

**Test Environment:**
OS: Windows 11 Version Dev (OS Build 22504.1010)
App: Windows Terminal Preview
Screen Reader: Narrator

### Steps to reproduce

**Repro Steps:**
1. Open Windows Terminal.
2. Open Narrator using 'Win + Ctrl + Enter' keys.
3. Press 'Ctrl + Shift + space' keys to open new tab Dropdown pop-up, Observe the issues.
4. Press 'Ctrl + Shift + r' keys to open Set tab color pop-up, Observe the issues.

**User Experience:**
Screen Reader users UX will not be good as they will not be able to know the purpose pop-ups.

**Guideline Reference:**
https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships

**Attachments:**

### Expected Behavior

Screen Reader announces Pop-up name when its focus first jump inside 'New Tab Dropdown' & 'Set tab color' pop-up.

### Actual Behavior

Screen Reader does not announce Pop-up name when its focus first jumps inside 'New Tab Dropdown' & 'Set tab color' pop-up.

**Observation:** Same issue repro with NVDA and JAWS Screen Reader.

**Comment by carlos-zamora** [comment]
In TerminalPage.cpp, if we just give the `newTabFlyout` an AutoProp name, we might be good?
